motionplan/tpSpaceRRT.go

```diff
@@ -302,62 +305,103 @@ func (mp *tpSpaceRRTMotionPlanner) getExtensionCandidate(
 	return cand
 }
 
-// extendMap grows the rrt map to the best candidate node if it is valid to do so.
+// attemptExtension will attempt to extend the rrt map towards the goal node, and will return the candidate added to the map that is
+// closest to that goal.
+func (mp *tpSpaceRRTMotionPlanner) attemptExtension(
+	ctx context.Context,
+	seedNode,
+	goalNode node,
+	rrt *rrtParallelPlannerShared,
+) node {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+		candidates := []*candidate{}
+		for ptgNum, curPtg := range mp.tpFrame.PTGs() {
+			// Find the best traj point for each traj family, and store for later comparison
+			// TODO: run in parallel
+			cand := mp.getExtensionCandidate(ctx, goalNode, ptgNum, curPtg, rrt, seedNode, mp.planOpts)
```

Comment:
Why not return an error separate from the candidate? Also, if there is an error we seem to just continue on is that what we want?